src/worker/vm/extensions/console.ts

```diff
@@ -1,11 +1,34 @@
 import { ConsoleExtension } from '@posthog/plugin-scaffold'
 
-export function createConsole(): ConsoleExtension {
+import { status } from '../../../shared/status'
+import { determineNodeEnv, NodeEnv, pluginDigest } from '../../../shared/utils'
+import { PluginConfig, PluginLogEntryType, PluginsServer } from '../../../types'
+
+export function createConsole(server: PluginsServer, pluginConfig: PluginConfig): ConsoleExtension {
+    async function consolePersist(type: PluginLogEntryType, ...args: unknown[]): Promise<void> {
+        if (determineNodeEnv() == NodeEnv.Development) {
+            status.info('👉', `${type} in ${pluginDigest(pluginConfig.plugin!, pluginConfig.team_id)}:`, ...args)
+        }
+
+        if (!server.ENABLE_PERSISTENT_CONSOLE) {
+            return
+        }
+
+        const message = args
+            .map((arg) => {
+                const argString = String(arg)
+                return argString === '[object Object]' ? JSON.stringify(arg, null, 4) : argString
```

Comment:
Being nitpicky here again but now this will lead to arrays being represented without brackets, which can be confusing:

```
> String([1,2,3]) // "1,2,3"
> JSON.stringify([1,2,3]) // "[1,2,3]"
```

Reply:
:+1: Addressed this case.